### PR TITLE
plan: fix a bug in `tryToGetMemTask` function

### PR DESCRIPTION
**Pull request by tiancaiamao**
isDistReq := !memDB && client != nil && client.IsRequestTypeSupported(kv.ReqTypeSelect, 0)

isDistReq has nothing to do with memDB, when memDB is false, that function should return.

@winoros @shenli @coocood

### CODE REVIEW

File: plan/physical_plan_builder.go [hunk 1]
```diff
@@ -124,7 +124,11 @@ func (ds *DataSource) tryToGetMemTask(prop *requiredProp) (task task, err error)
 	}
 	client := ds.ctx.GetClient()
 	memDB := infoschema.IsMemoryDB(ds.DBName.L)
-	isDistReq := !memDB && client != nil && client.IsRequestTypeSupported(kv.ReqTypeSelect, 0)
+	if !memDB {
+		return nil, nil
+	}
+
+	isDistReq := client != nil && client.IsRequestTypeSupported(kv.ReqTypeSelect, 0)
```
Comment (coocood) [inline plan/physical_plan_builder.go]: Why do we need to check isDistReq?
For memDB, we can just return mem table.

**Comment by coocood** [comment]
LGTM

File: plan/physical_plan_test.go [hunk 2]
```diff
@@ -1171,3 +1171,41 @@ func (s *testPlanSuite) TestAggEliminater(c *C) {
...
+	_, err = se.Execute(context.Background(), "use test")
+	c.Assert(err, IsNil)
+
+	sql := "select * from t where a in (1, 10, 20)"
+	expect := "TableReader(Table(t))->Sel([in(test.t.a, 1, 10, 20)])"
+
+	stmt, err := s.ParseOneStmt(sql, "", "")
+	c.Assert(err, IsNil)
+	sc := se.(sessionctx.Context).GetSessionVars().StmtCtx
+	sc.IgnoreTruncate = false
```
Comment (ngaut) [inline plan/physical_plan_test.go]: Why we need to set IgnoreTruncate ?

**Pull request merged**